### PR TITLE
ARROW-1954: [Python] Add metadata accessor to pyarrow.Field

### PR DESCRIPTION
**Pull request by kszucs**
Already implemented.

### CODE REVIEW

File: python/pyarrow/tests/test_types.py [hunk 1]
```diff
@@ -330,3 +343,35 @@ def test_field_equality_operators():
...
+
+
+def test_field_add_remove_metadata():
+    f0 = pa.field('foo', pa.int32())
+
+    assert f0.metadata is None
+
+    metadata = {b'foo': b'bar', b'pandas': b'badger'}
+
+    f1 = f0.add_metadata(metadata)
```
Comment (kszucs) [inline python/pyarrow/tests/test_types.py]: `add_metadata` is a bit misleading. It actually sets the metadata instead of adding new pairs to it.

Comment (xhochy) [reply python/pyarrow/tests/test_types.py]: Then we should maybe deprecate this method and replace with `replace_metadata`, `set_metadata` or `with_metadata`. Knowing Spark's immutable API, I would have chosen `with_metadata` but I guess this is probably not what the average Python user would expect as naming.

Comment (kszucs) [reply python/pyarrow/tests/test_types.py]: Both in tha Python and C++ api? Created issue https://issues.apache.org/jira/browse/ARROW-2769

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**